tests/unit/test_util.cpp

```diff
@@ -78,3 +78,37 @@ TEST_F(UtilTest, testFromHex)
 
     ASSERT_THROW(result = mococrw::utility::fromHex("  deadbeef"), mococrw::MoCOCrWException);
 }
+
+TEST_F(UtilTest, stringCleanse)
+{
+    std::string secret{"1234"};
+    mococrw::utility::stringCleanse(secret);
+    ASSERT_THAT(secret, Eq(std::string()));
+}
+
+TEST_F(UtilTest, stringCleanseCheckMemory)
+{
+    std::string secret{"1234"};
+    auto secretSize = secret.size();
+    std::string zeroedOutString(secretSize, '\0');
+    mococrw::utility::stringCleanse(secret);
+    ASSERT_GE(secret.capacity(), secretSize);
+    ASSERT_THAT(0, memcmp(&secret[0], &zeroedOutString[0], secretSize));
+}
+
+TEST_F(UtilTest, vectorCleanse)
+{
+    std::vector<uint8_t> secret = {222, 173, 190, 239};
+    mococrw::utility::vectorCleanse(secret);
+    ASSERT_THAT(secret, Eq(std::vector<uint8_t>()));
```

Comment:
This looks like a really complex way to test if secret.size() == 0 to me?

Reply:
Done